app/models/participant_declaration.rb

```diff
@@ -23,12 +22,13 @@ class ParticipantDeclaration < ApplicationRecord
   # Helper scopes
   scope :for_lead_provider, ->(cpd_lead_provider) { where(cpd_lead_provider: cpd_lead_provider) }
   scope :for_declaration, ->(declaration_type) { where(declaration_type: declaration_type) }
-  scope :for_profile, ->(profile) { joins(:profile_declaration).joins(:participant_profile).where(participant_profile: profile) }
+  scope :for_profile, ->(profile) { where(participant_profile: profile) }
   scope :started, -> { for_declaration("started").order(declaration_date: "desc").unique_id }
-  scope :uplift, -> { joins(:profile_declaration).merge(ProfileDeclaration.uplift) }
-  scope :ect, -> { joins(:profile_declaration).merge(ProfileDeclaration.ect_profiles) }
-  scope :mentor, -> { joins(:profile_declaration).merge(ProfileDeclaration.mentor_profiles) }
-  scope :npq, -> { joins(:profile_declaration).merge(ProfileDeclaration.npq_profiles) }
+
+  scope :uplift, -> { where(participant_profile_id: ParticipantProfile.uplift.select(:id)) }
+  scope :ect, -> { where(participant_profile_id: ParticipantProfile::ECT.select(:id)) }
+  scope :mentor, -> { where(participant_profile_id: ParticipantProfile::Mentor.select(:id)) }
+  scope :npq, -> { where(participant_profile_id: ParticipantProfile::NPQ.select(:id)) }
```

Comment:
Are these like `{ .joins(:participant_profile).merge(ParticipantProfile.uplift) }`? And so on for other scopes?

Reply:
Similar. Stan changes them to this as a sub-query. That's where the joins discussion vs includes that I alluded to came from a few months back.